File: codewarcpp/reducebutgrow.cpp
#include <vector>
#include <numeric>
#include <functional>
#include <iostream>

int grow(const std::vector<int>& nums) {
  return std::accumulate(nums.cbegin(), nums.cend(), 1, std::multiplies<int>());
}

int grow(std::vector<int> nums) {
  int result = 1;
  for(auto x: nums)  result = result * x;
  return result;
}

File: codwar.go/reducebutgrow.go
package main

func Grow(arr []int) int {
	result := 1

	for _, n := range arr {
		result *= n
}
	return result
}

func main() {
	println(Grow([]int{1, 2, 3, 4, 5, 6}) == 720)
	println(Grow([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}) == 3628800)
	println(Grow([]int{1, 2, 3, 4, 5},),120)
	println(Grow([]int{}) == 1)
}
